blender: 4.3.2 -> 4.4.0

amarshall opened the pull request:
See individual commits for details.

I’m on the fence about making the OpenCOLLADA Blender fork top-level, especially as Blender expects to remove it in v5.0 later this year anyway. Thoughts welcome.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

amarshall committed:
opencollada-blender: init at unstable-2025-01-30

Blender is now using a fork of OpenCOLLADA. They have done so since 4.2,
but in the upcoming Blender 4.4, Blender will fail to build with the
original and needs this fork. This fork “removes everything (a lot!)
that is not actually needed by Blender” and so is likely not suitable
for use by other projects (of which there are a few within Nixpkgs). See
more in [upstream README][1].

[1]: https://github.com/aras-p/OpenCOLLADA/blob/4526eb8aaa6462c71fbedd23103976c151a01c50/README.md

amarshall committed:
blender: move to pkgs/by-name

amarshall committed:
blender: 4.3.2 -> 4.4.0

Changelog: https://developer.blender.org/docs/release_notes/4.4/

Diff: https://projects.blender.org/blender/blender/compare/v4.3.2..v4.4.0

- Blender now needs its fork of OpenCOLLADA to build successfully
- Blender is now bundling assets with the release tarballs again, so go
  back to single src
- Blender changed Draco location so we can have it work out-of-the-box
  and don’t have to move the Python dir around anymore; see upstream
  faaa12a10131b93757a8395f3448040436981e5f
- Other patches we backported are now applied and unneeded
- Blender added Pipewire support, but it’s off in their release builds
  currently so turn it off as well

VuiMuich commented:
The package builds fine, but I found some visual glitches (on switching workspace occasionally some icon gets drawn really big possible at the mouse cursor position for a split second, and on the sculpt workspace the icon for brushes are messed up) and one crash when trying to assign a texture to the default cube to test texture painting. (Can post the stdout later, when back on my machine).

VuiMuich commented on pkgs/by-name/bl/blender/package.nix:
```diff
@@ -201,6 +172,7 @@ stdenv'.mkDerivation (finalAttrs: {
       "-DWITH_OPENIMAGEDENOISE=${if openImageDenoiseSupport then "ON" else "OFF"}"
       "-DWITH_OPENSUBDIV=ON"
       "-DWITH_OPENVDB=ON"
+      "-DWITH_PIPEWIRE=OFF"
```
Is there a specific reason, why we want no pipewire?
Or could we make this optional, same as jack?

VuiMuich replied:
Just saw your not in the commit message, sorry for not looking there first.

Still an option might be nice to have, no?

VuiMuich replied:
I see, with that It sure makes to wait for upstream defaulting to `ON`.

VuiMuich commented on pkgs/by-name/bl/blender/package.nix:
```diff
@@ -114,42 +112,15 @@ in
-      hash = "sha256-LCU2JpQbvQ+W/jC+H8J2suh+X5sTLOG9TcE2EeHqVh4=";
-    })
-    (fetchgit {
-      name = "assets";
```
Oops, missed that.

VuiMuich commented:
So, both visual glitches happen as well on my sons machine with Garuda Linux, also he is on Wayland, while I am running X11. I'll see if there is an upstream issue for this.

Furthermore, I wasn't able to reproduce the crash as well.

VuiMuich commented:
Yes, that's exactly the glitch I am seeing on the Sculpting workspace.

I also found an additional detail in the flashing icon glitch: it is the last icon of the tool menu to the left. When the tools are hidden (e.g. by pressing `T`) it does not appear.

I can confirm both glitches on `4.3.2` and `4.4.0`, on Wayland and X11, on NixOS and on Garuda, on system packages and the official binary package from blender.org. But neither macOS nor Windows seem to be affected.
I only had a brief look at upstream issues and could not find something, so if tomorrow is a rainy day as predicted, I'll open upstream issues for both.

veprbl commented on pkgs/by-name/op/opencollada-blender/package.nix:
```diff
@@ -0,0 +1,45 @@
+{
```
Is it possible to implement this as an override?

veprbl replied:
This is clear now. Thank you.

veprbl commented on pkgs/by-name/op/opencollada-blender/package.nix:
```diff
@@ -0,0 +1,45 @@
+
+stdenv.mkDerivation {
+  pname = "opencollada";
+
```
```suggestion
```

veprbl commented on pkgs/by-name/op/opencollada-blender/package.nix:
```diff
@@ -0,0 +1,45 @@
+    pkg-config
+  ];
+
+  buildInputs = lib.optionals stdenv.hostPlatform.isDarwin (
+    with darwin.apple_sdk.frameworks; [ AGL ]
+  );
+
```
```suggestion
```
Our darwin SDK has been reworked, so these frameworks are not needed and provide empty dummies.

amarshall committed:
opencollada: formatting

Alphabetize params, reorder attrs to be more idiomatic, adjust spacing.

amarshall committed:
opencollada: remove unneeded Apple SDK

Per PR review:

> Our darwin SDK has been reworked, so these frameworks are not needed
> and provide empty dummies.
